packages/core/src/common/messaging/web-socket-channel.ts

```diff
@@ -112,5 +113,130 @@ export interface IWebSocket {
      * @param cb The callback.
      */
     onClose(cb: (reason: string, code?: number) => void): void;
+    /**
+     * Listener callback to handle connect events (Remote side).
+     * @param cb The callback.
+     */
+    onConnect(cb: () => void): void;
 }
 
+/**
+ * An persist websocket implementation base on the `{@link IWebSocket} .
+ * For the client side, it will always try reconnect to remote .
+ * For the sever side, it will fire close state after the `ReconnectionGraceTime` time
+ */
+export class PersistentWebSocket implements IWebSocket {
```

Comment:
```suggestion
/**
 * A persistant WebSocket implementation based on the `{@link IWebSocket}.
 * For the client side, it will always try to reconnect to the remote.
 * For the sever side, it will close itself after the `ReconnectionGraceTime` time.
 */
export class PersistentWebSocket implements IWebSocket {
```